DependencyInjection/SonataUserExtension.php

```diff
@@ -273,4 +257,55 @@ protected function aliasManagers(ContainerBuilder $container, $managerType)
         $container->setAlias('sonata.user.user_manager', sprintf('sonata.user.%s.user_manager', $managerType));
         $container->setAlias('sonata.user.group_manager', sprintf('sonata.user.%s.group_manager', $managerType));
     }
+
+    /**
+     * @param array $config
+     */
+    private function checkManagerTypeToModelTypeMapping(array $config)
+    {
+        $managerType = $config['manager_type'];
+
+        $actualModelClasses = [
+            $config['class']['user'],
+            $config['class']['group'],
+            $config['admin']['user']['class'],
+            $config['admin']['group']['class'],
+        ];
+
+        if ('orm' === $managerType) {
+            $expectedModelClasses = [
+                EntityUser::class,
+                EntityGroup::class,
+                EntityUserAdmin::class,
+                EntityGroupAdmin::class,
+            ];
+        } elseif ('mongodb' === $managerType) {
+            $expectedModelClasses = [
+                DocumentUser::class,
+                DocumentGroup::class,
+                DocumentUserAdmin::class,
+                DocumentGroupAdmin::class,
+            ];
+        } else {
+            throw new \InvalidArgumentException(sprintf('Invalid manager type "%s".', $managerType));
```

Comment:
It would be great to drop `sprintf` usage for string encapsulation instead. This is about performance.

- https://blog.blackfire.io/php-7-performance-improvements-encapsed-strings-optimization.html
- https://jonczyk.me/2016/10/23/phpyths-buster-great-string-performance-test/

Maybe the rules should be updated on dev-kit.

Reply:
Personally, that's typically the type of string where I'd stick with `sprintf`. It's not about performance, it's about legibility, especially when you have double quotes like here.

Reply:
and so you really need performance when you misconfigure sth in dev mode?

I would like to stick with the sprintf()

Reply:
Well, let's go for it.